packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -595,6 +635,15 @@ export interface GitHubEnterpriseSourceProps extends ThirdPartyGitSourceProps {
    * @default false
    */
   readonly ignoreSslErrors?: boolean;
+
+  /**
+   * The commit ID, pull request ID, branch name, or tag name that corresponds to
+   * the version of the source code you want to build
+   *
+   * @example 'pr/25'
+   * @default the default branch's HEAD commit ID is used
+   */
+  readonly sourceVersion?: string;
```

Comment:
Are we 100% sure this is supported for `GitHubEnterprise` Sources?

Reply:
Actually, here's an idea.

If it is supported for GitHub Enterprise, then we have a common parent of all Git-backed sources in this file, called `GitSource`. You can just add `brachOrRef` as a property there, instead of duplicating this code in every subclass!

Reply:
[This page](https://docs.aws.amazon.com/codebuild/latest/userguide/sample-source-version.html) seems to confirm that it is supported with `GithubEnterprise` sources.

Love that idea. I added `branchOrRef` to `GitSource` and removed it from every subclass, as you suggested.